include/ebpf_structs.h

```diff
@@ -313,6 +313,11 @@ struct bpf_link_info
         {
             uint32_t ifindex;
         } xdp;
+        struct
+        {
+            uint64_t cgroup_id;
+        } cgroup;
```

Comment:
libbpf / bpftool uses bpf_link_type values to say which union member is used, with
BPF_LINK_TYPE_XDP for xdp and BPF_LINK_TYPE_CGROUP for cgroup.  For consistency, I'd suggest we do that instead of using the same value for both.

Reply:
I dont understand this feedback. Are you talking about the `uint8_t attach_data;` arm of the union?

Reply:
No:
```
   enum bpf_link_type type;               ///< Link type.
```
The above field says which arm of the union is present.   We currently have:
```
// The link type is used to tell which union member is present
// in the bpf_link_info struct.  There is exactly one non-zero value
// per union member.
enum bpf_link_type
{
    BPF_LINK_TYPE_UNSPEC, ///< Unspecified link type.
    BPF_LINK_TYPE_PLAIN,  ///< Normal link type.
};
```

Linux has:
```
enum bpf_link_type {
    BPF_LINK_TYPE_UNSPEC = 0,
...
    BPF_LINK_TYPE_CGROUP = 3,
...
    BPF_LINK_TYPE_XDP = 6,
...
};
```
Where the "cgroup", "xdp", etc. match the union member name.

Reply:
I have expanded the `bpf_link_type` enum and I am populating the `link_info` with the `link_type`. However, when copying the attach parameter, I am not using the `link_info` as I would prefer the EC code to be agnostic of various eBPF program attach types.